listener.go

```diff
@@ -133,31 +133,14 @@ func (l *listener) Close() error {
 	l.closed = true
 	err := l.Listener.Close()
 
-	// Drain nextConn
-drainNextConn:
-	for {
-		select {
-		case conn := <-l.nextConn:
-			conn.Close()
-		default:
-			break drainNextConn
-		}
-	}
 	close(l.nextConn)
+	for conn := range l.nextConn {
+		conn.Close()
+	}
 
-	// Drain nextErr
-drainNextErr:
-	for {
-		select {
-		case e := <-l.nextErr:
-			if e == nil { // it may already be closed
-				break drainNextErr
-			} else {
-				err = e
-			}
-		default:
-			break drainNextErr
-		}
+	// l.nextErr is always closed at this point
```

Comment:
This is a little misleading I think.  It's closed when listen() exits, so it's sometimes closed at this point -- here we're waiting for it to close and trying to unblock listen() so it can exit (eg if it's trying to write to nextErr and nobody is calling Accept())